mteb/tasks/Retrieval/ara/SadeemKeywordRetrieval.py

```diff
@@ -0,0 +1,63 @@
+from __future__ import annotations
+
+import datasets
+
+from mteb.abstasks.TaskMetadata import TaskMetadata
+
+from ....abstasks.AbsTaskRetrieval import AbsTaskRetrieval
+
+
+class SadeemKeywordRetrieval(AbsTaskRetrieval):
+    _EVAL_SPLIT = "test"
+
+    metadata = TaskMetadata(
+        name="SadeemKeywordRetrieval",
+        dataset={
+            "path": "sadeem-ai/sadeem-ar-eval-retrieval-keywords",
+            "revision": "6e8b967b0fc44f4cebb34a8cfab0eb7fd9484783",
+            "name": "test",
+        },
+        reference="https://huggingface.co/datasets/sadeem-ai/sadeem-ar-eval-retrieval-keywords",
+        description="SadeemKeyword: A Benchmark Data Set for Community Keyword-Retrieval Research",
+        type="Retrieval",
+        category="s2p",
+        eval_splits=[_EVAL_SPLIT],
+        eval_langs=["ara-Arab"],
+        main_score="ndcg_at_10",
+        date=("2024-01-01", "2024-04-01"),
+        form=["written"],
+        domains=["written"],
+        task_subtypes=["Article retrieval"],
+        license="Not specified",
+        socioeconomic_status="medium",
+        annotations_creators="derived",
+        dialect=[],
+        text_creation="found",
+        bibtex_citation="""
+            @inproceedings{sadeem-2024-ar-retrieval-keywords,
+            title = "{QMS}um: A New Benchmark for Arabic Keywoord-based Articles Searching.",
+            author = "Abu Bakr Soliman | Saddem-ai"
+            }
+        """,
+        n_samples={_EVAL_SPLIT: 22979},
```

Comment:
Seems like quite a large dataset any chance we can downsample this?

Reply:
Hi
Downsampling just the number, or the whole dataset?

Reply:
Not sure what is meant by number here, but I would downsample the number of documents needed to be embedded.

Reply:
@KennethEnevoldsen 
I was talking about the configuration value of `_EVAL_SPLIT`

I already uploaded a previous dataset to `mmteb` (not my first contribution) with the same amount of samples, but for different evaluation case.

Also, you have many larger datasets. ([example](https://huggingface.co/datasets/mteb/cqadupstack-english))

This's a developed dataset we used in our company for a while, and we decided to make it public.
 
I am not sure that we can develop a new one now.

I hope this will not prevent accepting the PR.

Reply:
Yes, MMTEB does contain many large datasets, but there has been an effort to speed it up and limit excessively large datasets (see e.g. #638, #381, #481, as well as the checkbox above). Especially since large datasets limit how many people can use the MTEB benchmark due to compute costs.

What I am asking here is not that you develop a new dataset, but that before adding it in might be ideal to downsample it. My guess (which we have also seen from previous submissions) is that for retrieval it does inflate the performance scores slightly to reduce the corpus size, but it does not change the ranking between models.

Reply:
Yes. We will work to update it this week

Reply:
@KennethEnevoldsen 
We updated the dataset. It's now only 7179 samples.